web/src/main/java/org/cbioportal/web/StudyViewController.java

```diff
@@ -501,34 +476,65 @@
+                            } else if (!(molecularProfile.getMolecularAlterationType()
+                                .equals(MolecularProfile.MolecularAlterationType.STRUCTURAL_VARIANT)
+                                && molecularProfile.getDatatype().equals("SV"))) {
+                                // replace structural variant profile with
+                                // mutation profile having fusion data
+                                structuralVariantProfile= molecularProfile;
+                            }
+                        } else {
+                            profilesToReturn.add(molecularProfile);
+                        }
+                    }
+
+                    if (structuralVariantProfile != null) {
+                        profilesToReturn.add(structuralVariantProfile);
+                    }
+
+                    return profilesToReturn;
+            }));
+
+        List<MolecularProfileCaseIdentifier> molecularProfileSampleIdentifiers = new ArrayList<>();
 
-        List<String> queryMolecularProfileIds = new ArrayList<>();
-        List<String> querySampleIds = new ArrayList<>();
         for (int i = 0; i < studyIds.size(); i++) {
             String studyId = studyIds.get(i);
             String sampleId = sampleIds.get(i);
             if (studyMolecularProfilesSet.containsKey(studyId)) {
                 studyMolecularProfilesSet.get(studyId).stream().forEach(molecularProfile -> {
-                    queryMolecularProfileIds.add(molecularProfile.getStableId());
-                    querySampleIds.add(sampleId);
+                    MolecularProfileCaseIdentifier profileCaseIdentifier = new MolecularProfileCaseIdentifier();
+                    profileCaseIdentifier.setMolecularProfileId(molecularProfile.getStableId());
+                    profileCaseIdentifier.setCaseId(sampleId);
+                    molecularProfileSampleIdentifiers.add(profileCaseIdentifier);
                 });
             }
         }
 
         List<GenePanelData> genePanelData = genePanelService
-                .fetchGenePanelDataInMultipleMolecularProfiles(queryMolecularProfileIds, querySampleIds);
-        HashMap<String, Integer> molecularPorfileSampleCountSet = new HashMap<String, Integer>();
+                .fetchGenePanelDataInMultipleMolecularProfiles(molecularProfileSampleIdentifiers);
+        HashMap<String, Integer> molecularProfileSampleCountSet = new HashMap<>();
 
         for (GenePanelData datum : genePanelData) {
             if (datum.getProfiled()) {
-                Integer count = molecularPorfileSampleCountSet.getOrDefault(datum.getMolecularProfileId(), 0);
-                molecularPorfileSampleCountSet.put(datum.getMolecularProfileId(), count + 1);
+                Integer count = molecularProfileSampleCountSet.getOrDefault(datum.getMolecularProfileId(), 0);
+                molecularProfileSampleCountSet.put(datum.getMolecularProfileId(), count + 1);
             }
         }
```

Comment:
This could be done with a groupby. Might be a bit easier to read, I'll leave it up to you.

```Java
Map<String, Integer> molecularProfileSampleCountSet = genePanelData.stream()
    .filter(GenePanelData::getProfiled)
    .collect(Collectors.groupingBy(GenePanelData::getMolecularProfileId, Collectors.summingInt(s -> 1)));
```

Or, if you can accept a long, you could simplify it further

```Java
Map<String, Long> molecularProfileSampleCountSet = genePanelData.stream()
    .filter(GenePanelData::getProfiled)
    .collect(Collectors.groupingBy(GenePanelData::getMolecularProfileId, Collectors.counting()));
```